File: 2300-successful-pairs-of-spells-and-potions/2300-successful-pairs-of-spells-and-potions.cs
public class Solution {
    public int[] SuccessfulPairs(int[] s, int[] po, long t) {
        Array.Sort(po);
        for(int i=0;i<s.Length;i++){
            int x=s[i],ans=po.Length;
            int st=0,en=po.Length-1;
            while(st<=en){
                int m= (st+en)/2;
                long y = (long)x * po[m];
                if(y>=t){
                    ans=m;
                    en=m-1;
                }
                else{
                    st=m+1;
                }
            }
            s[i]= po.Length - ans;
        }
        return s;
        
    }
}

File: 2300-successful-pairs-of-spells-and-potions/2300-successful-pairs-of-spells-and-potions.cpp
class Solution {
public:
    vector<int> successfulPairs(vector<int>& s, vector<int>& p, long long t) {
        sort(p.begin(),p.end());
        for(int i=0;i<s.size();i++){
            s[i]=help(p,s[i],t);
        }
        return s;
    }
    int help(vector<int>& p,long long x,int target){
        int n = p.size();
        int low = 0, high = n - 1, ans = n;

        while (low <= high) {
            int mid = low + (high - low) / 2;
            long long val = x * p[mid];

            if (val >= target) {
                ans = mid;       
                high = mid - 1;  
            } else {
                low = mid + 1;
            }
        }

        return n-ans;
    }
};